api/admin/controller/patron_auth_service_self_tests.py

```diff
@@ -2,116 +2,54 @@
 
 from typing import Any, Dict, Optional, Type
 
-import flask
 from flask import Response
 from sqlalchemy.orm import Session
 
-from api.admin.problem_details import *
-from api.authentication.base import AuthenticationProvider
+from api.admin.controller.self_tests import IntegrationSelfTestsController
+from api.admin.problem_details import FAILED_TO_RUN_SELF_TESTS
+from api.authentication.base import AuthenticationProviderType
 from api.integration.registry.patron_auth import PatronAuthRegistry
-from core.integration.goals import Goals
 from core.integration.registry import IntegrationRegistry
-from core.model import get_one, json_serializer
-from core.model.integration import (
-    IntegrationConfiguration,
-    IntegrationLibraryConfiguration,
-)
+from core.model.integration import IntegrationConfiguration
 from core.util.problem_detail import ProblemDetail, ProblemError
 
 
-class PatronAuthServiceSelfTestsController:
+class PatronAuthServiceSelfTestsController(
+    IntegrationSelfTestsController[AuthenticationProviderType]
```

Comment:
Much of what used to exist in this class has been moved to `IntegrationSelfTestsController`, so it can be shared with `CollectionSelfTestsController`.